alternator/executor.cc

```diff
@@ -459,12 +465,22 @@ static rjson::value fill_table_description(schema_ptr schema, table_status tbl_s
     // when the table was created. But, Spark jobs expect something to be returned
     // and PAY_PER_REQUEST seems closer to reality than PROVISIONED.
     rjson::add(table_description, "BillingModeSummary", rjson::empty_object());
-    rjson::add(table_description["BillingModeSummary"], "BillingMode", "PAY_PER_REQUEST");
+    if (tags_ptr && tags_ptr->contains(RCU_TAG_KEY)) {
+        rjson::add(table_description["BillingModeSummary"], "BillingMode", "PROVISIONED");
+    } else {
+        rjson::add(table_description["BillingModeSummary"], "BillingMode", "PAY_PER_REQUEST");
+    }
+
     rjson::add(table_description["BillingModeSummary"], "LastUpdateToPayPerRequestDateTime", rjson::value(creation_date_seconds));
     // In PAY_PER_REQUEST billing mode, provisioned capacity should return 0
     rjson::add(table_description, "ProvisionedThroughput", rjson::empty_object());
-    rjson::add(table_description["ProvisionedThroughput"], "ReadCapacityUnits", 0);
-    rjson::add(table_description["ProvisionedThroughput"], "WriteCapacityUnits", 0);
+    if (tags_ptr && tags_ptr->contains(RCU_TAG_KEY)) {
+        rjson::add(table_description["ProvisionedThroughput"], "ReadCapacityUnits", std::stoi(tags_ptr->at(RCU_TAG_KEY)));
+        rjson::add(table_description["ProvisionedThroughput"], "WriteCapacityUnits", std::stoi(tags_ptr->at(WCU_TAG_KEY)));
```

Comment:
Please note that users currently can write to "system:"-prefixed tags directly (this is useful, e.g., for setting write isolation or initial_tablets) so you can get something surprising like rcu set but wcu unset. I see you always check only if rcu is set, perhaps you should check both of them, otherwise this `at` can throw.
Also, std::stoi() can also throw if the value is not an integer...

Perhaps can do something like:
```c++
int rcu = 0;
int wcu = 0;
if (tags_ptr) {
    auto rcu_tag = tags_ptr.find(RCU_TAG_KEY);
    auto wcu_tag = tags_ptr.find(WCU_TAG_KEY);
    if (rcu_tag && wcu_tag) {
          // do atoi() and ignore failure (leaving rcu and wcu as 0)
          ...
          rcu = ..
          wcu = ...
```